src/train.py

```diff
@@ -0,0 +1,122 @@
+    train_encoder_states = data.load_encoder_states(args, "train")
+    training_dataset = dataset.CombinedDataset(train_encoder_states, explanation_dataset['train'], args.layer)
+    validation_dataset = dataset.CombinedDataset(data.load_encoder_states(args, "validation"), explanation_dataset['validation'], args.layer)
+    dataloader = DataLoader(training_dataset, batch_size=args.batch_size, collate_fn=collate_fn_decode)
+    model = lstm_probing(input_size=len(tokenizer), hidden_size=train_encoder_states.shape[2], lstm_hidden_size=args.lstm_hidden_size,
+    device=args.device, output_size=len(tokenizer))
+    model = model.to(args.device)
+    optimizer = optim.Adam(model.parameters(), lr=args.learning_rate)
+    criterion = nn.CrossEntropyLoss()
+    wandb.watch(model, criterion, log="all")
+    early_stopper = util.EarlyStopper(patience=3, min_delta=0)
+    print_loss_total = 0 
+    plot_loss_total = 0 
+    start = time.time()
+    best_bleu = 0
+
+    for epoch in range(1, args.max_epochs + 1):
+        loss, perplexity = train(model, dataloader, criterion, epoch, optimizer, args)
+        print_loss_total += loss
+        plot_loss_total += loss
+
+        if epoch % args.evaluate_every == 0:
+            decoded_words, reference_words, bleu_score, loss, perplexity = evaluate(tokenizer, model, validation_dataset, criterion, args)
+            avg_bleu_score = bleu_score / len(explanation_dataset['validation'])
+            wandb.log({'eval_epoch': epoch, 'eval_loss': loss, 'eval_perplexity': perplexity, 'bleu_score': avg_bleu_score})
+            print({'eval_epoch': epoch, 'eval_loss': loss, 'eval_perplexity': perplexity, 'bleu_score': avg_bleu_score})
+            if avg_bleu_score > best_bleu:
+                torch.save(model.state_dict(), 'train-{date:%Y-%m-%d_%H}'.format( date=datetime.datetime.now()))
+                best_bleu = avg_bleu_score
+                name = 'validation' + '-{date:%Y-%m-%d_%H}'.format( date=datetime.datetime.now())
+                #np.savetxt('../results/'+name+'_decoded_words.txt', decoded_words, fmt='%s')
+                decoded_words_output = open('../results/'+name+'_decoded_words.pkl', 'wb')
+                pickle.dump(decoded_words, decoded_words_output)
+                decoded_words_output.close()
+            if early_stopper.early_stop(avg_bleu_score):             
+                break
+
+        if epoch % args.print_every == 0:
+            print_loss_avg = print_loss_total / args.print_every
+            print_loss_total = 0
+            print('%s (%d %d%%) %.4f' % (util.timeSince(start, epoch / args.max_epochs),
+                                         epoch, epoch / args.max_epochs * 100, print_loss_avg))
+    
+    return model, dataloader
+
+
+def train(model, dataloader, criterion, epoch, optimizer, args):
+    model.train()
+    # x dimension: (batch, 1, seq_lenth=768)
+    # y dimension: (batch, seq_lenth=55)
+    for batch, (x, y, y_2, y_3) in enumerate(tqdm(dataloader)):
```

Comment:
It seems like y_2 and y_3 aren't used at all in training, correct? Should this be changed to (x, y, _, _) to indicate they aren't actually used?